dvc/remote/s3.py

```diff
@@ -40,29 +40,22 @@ class RemoteS3(RemoteBase):
     PARAM_ETAG = 'etag'
 
     def __init__(self, project, config):
-        import configobj
-
         self.project = project
-        storagepath = 's3://' + config.get(Config.SECTION_AWS_STORAGEPATH,
-                                           '').lstrip('/')
+
+        storagepath = 's3://{}'.format(
+            config.get(Config.SECTION_AWS_STORAGEPATH, '').lstrip('/')
+        )
+
         self.url = config.get(Config.SECTION_REMOTE_URL, storagepath)
-        self.region = config.get(Config.SECTION_AWS_REGION, None)
-        self.profile = os.getenv('AWS_PROFILE',
-                                 config.get(Config.SECTION_AWS_PROFILE,
-                                            'default'))
-        self.endpoint_url = config.get(Config.SECTION_AWS_ENDPOINT_URL, None)
+        self.region = config.get(Config.SECTION_AWS_REGION)
+        self.profile = config.get(Config.SECTION_AWS_PROFILE)
+        self.endpoint_url = config.get(Config.SECTION_AWS_ENDPOINT_URL)
 
-        credentialpath = config.get(Config.SECTION_AWS_CREDENTIALPATH, None)
-        if credentialpath:
-            creds_conf = configobj.ConfigObj(credentialpath)
-            creds = creds_conf.get(self.profile, {})
-        else:
-            creds = self._get_aws_config_from_env()
+        # backward compatibility
+        credentialpath = config.get(Config.SECTION_AWS_CREDENTIALPATH)
 
-        self.creds = creds
-        self.region = creds.get('region', self.region)
-        self.aws_access_key_id = creds.get('aws_access_key_id', None)
-        self.aws_secret_access_key = creds.get('aws_secret_access_key', None)
+        if credentialpath:
+            os.environ['AWS_SHARED_CREDENTIALS_FILE'] = credentialpath
```

Comment:
Should we override it only if it's not defined yet? It's a matter of priorities - our config path vs env var first. I would expect env var to go first.

Reply:
I didn't think about that case! I agree with pushing up `env vars` on the priority list, let me send a commit fixing that.